modules/vmfcore/src/zlib_compressor.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2015 Intel(r) Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http ://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "vmf/zlib_compressor.hpp"
+#include "zlib.h"
+
+namespace vmf {
+
+void ZLibCompressor::compress(const MetaString& input, vmf_rawbuffer& output)
+{
+    //TODO: (length of string)*(size of character)
+    size_t srcLen = input.length()*sizeof(MetaString::value_type);
+    size_t destBound = compressBound(srcLen);
+    size_t destLength = destBound;
+    //should also keep the size of source data
+    //for further decompression
+    u_char* destBuf = (u_char*)malloc(destBound+sizeof(size_t));
+    if(!destBuf)
+    {
+        VMF_EXCEPTION(InternalErrorException, "Out of memory");
+    }
+
+    *((size_t*)destBuf) = srcLen;
```

Comment:
use of `size_t` isn't portable - I suggest we use `vmf_integer` (that is `int64_t`) here!